src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -0,0 +1,646 @@
+    """Prep PPL for use in override output sheet pre-utility subgroups."""
+    logger.debug("Prepping Plant Parts Table")
+
+    # Add utilty name eia and only take relevant columns
+    ppl_out = (
+        ppl.reset_index()
+        .merge(
+            pudl_out.utils_eia860()[
+                ["utility_id_eia", "utility_name_eia", "report_date"]
+            ].copy(),
+            on=["utility_id_eia", "report_date"],
+            how="left",
+            validate="m:1",
+        )[RELEVANT_COLS_PPL]
+        .copy()
+    )
+
+    return ppl_out
+
+
+def _prep_deprish(deprish, pudl_out):
+    """Prep depreciation data for use in override output sheet pre-utility subgroups."""
+    logger.debug("Prepping Deprish Data")
+
+    # Get utility_id_eia from EIA
+    util_df = pudl_out.utils_eia860()
+    id_dict = dict(zip(util_df["utility_id_pudl"], util_df["utility_id_eia"]))
+
+    deprish_out = deprish.assign(
+        report_year=lambda x: x.report_date.dt.year.astype("Int64"),
+        utility_id_eia=lambda x: x.utility_id_pudl.map(id_dict).astype("Int64"),
+    )
+
+    return deprish_out
+
+
+def _generate_input_dfs(pudl_out, rmi_out):
+    """Load and prep and all the input tables."""
+    logger.debug("Generating inputs")
+    inputs_dict = {
+        "ferc_eia": rmi_out.ferc1_to_eia().pipe(_prep_ferc_eia, pudl_out),
+        "ppl": rmi_out.plant_parts_eia().pipe(_prep_ppl, pudl_out),
+        "deprish": rmi_out.deprish().pipe(_prep_deprish, pudl_out),
+    }
+
+    return inputs_dict
+
+
+def _get_util_year_subsets(inputs_dict, util_id_eia_list, years):
+    """Get utility and year subsets for each of the input dfs."""
+    util_year_subset_dict = {}
+    for df_name, df in inputs_dict.items():
+        logger.debug(f"Getting utility-year subset for {df_name}")
+        in_year_range = df["report_year"].isin(years)
+        in_eia_util_list = df["utility_id_eia"].isin(util_id_eia_list)
+        subset_df = df[in_year_range & in_eia_util_list].copy()
+        # Make sure dfs aren't too big...
+        assert (
+            len(subset_df) < 500000
+        ), "Your subset is more than 500,000 rows...this \
+            is going to make excel reaaalllllyyy slow. Try entering a smaller utility \
+            or year subset"
+
+        if df_name == "ferc_eia":
+            # Add column with excel formula to check if the override record id is the
+            # same as the AI assigend id. Doing this here instead of prep_ferc_eia
+            # because it is based on index number which is changes when you take a
+            # subset of the data.
+            subset_df = (
+                subset_df.reset_index()
+                .assign(
+                    used_match_record=lambda x: (  # can this be moved to prep?
+                        "=(J"
+                        + (x.index + 2).astype("str")
+                        + "=F"
+                        + (x.index + 2).astype("str")
+                        + ")"
```

Comment:
Ugh yes, it is sooo fragile I will work on this

Reply:
So the index I'm having trouble with is the row index rather than the column index. I can find the correct columns by using column name, but the formulas also require the row index (ex: J3 = F3, J4=F4....) where 3 and 4 are row numbers. When I do this with the whole ferc-eia table (before it gets split into utility subsets, the index values are tied to the whole table rather than the smaller tables where the index gets reset. The formulas won't work if I create them according to the index in the whole table.

Reply:
I left it in the get_util_year_subsets() function because of the index issue, but I cleaned it up a bit so the column references are better and it's vectorized.